ipc/api/src/checkpoint.rs

```diff
@@ -53,22 +57,27 @@ impl Display for QuorumReachedEvent {
 pub struct BottomUpCheckpointBundle {
     pub checkpoint: BottomUpCheckpoint,
     /// The list of signatures that have signed the checkpoint hash
+    #[serde(deserialize_with = "deserialize_vec_bytes_from_vec_hex")]
+    #[serde(serialize_with = "serialize_vec_bytes_to_vec_hex")]
     pub signatures: Vec<Signature>,
     /// The list of addresses that have signed the checkpoint hash
     pub signatories: Vec<Address>,
 }
 
 /// The collection of items for the bottom up checkpoint submission
+#[serde_as]
 #[derive(PartialEq, Eq, Clone, Debug, Serialize, Deserialize)]
 pub struct BottomUpMsgBatch {
     /// Child subnet ID, for replay protection from other subnets where the exact same validators operate
+    #[serde_as(as = "HumanReadable")]
     pub subnet_id: SubnetID,
     /// The height of the child subnet at which the batch was cut
     pub block_height: ChainEpoch,
     /// Batch of messages to execute
     pub msgs: Vec<IpcEnvelope>,
 }
 
+#[serde_as]
 #[derive(PartialEq, Eq, Clone, Debug, Serialize, Deserialize)]
```

Comment:
The title says "Adding display to bottom up checkpoint". Is `Display` supposed to be here?

Reply:
It started with adding display, then slowly grows into better presenting/logging checkpoint and its related structs. Naming is hard. I will change the title.